core/src/main/java/io/cucumber/core/exception/CompositeCucumberException.java

```diff
@@ -1,25 +1,32 @@
 package io.cucumber.core.exception;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
+
+public class CompositeCucumberException
+    extends CucumberException {
 
-public class CompositeCucumberException extends CucumberException {
     private final List<Throwable> causes;
 
-    public CompositeCucumberException(List<Throwable> causes) {
-        super(String.format("There were %d exceptions:", causes.size()));
-        this.causes = causes;
+    public CompositeCucumberException(final List<Throwable> causes) {
+        super(String.format("There were %d exceptions:",
+            Objects.isNull(causes) ? 0 : causes.size()
+        ));
+        this.causes = Objects.isNull(causes) ? new ArrayList<>() : causes;
     }
 
     public List<Throwable> getCauses() {
         return Collections.unmodifiableList(this.causes);
     }
 
     public String getMessage() {
-        StringBuilder sb = new StringBuilder(super.getMessage());
-        for (Throwable e : this.causes) {
+        final StringBuilder sb = new StringBuilder(super.getMessage());
+        for (final Throwable e : this.causes) {
```

Comment:
kept having issues with people reusing variables, and/or code merges, and due to variables being final the compiler errors instead of waiting for unit tests, or cucumber testing or manually regression testing so it's auto pilot for me. again will not add in cucumber code base